src/dstack/_internal/core/backends/datacrunch/compute.py

```diff
@@ -0,0 +1,139 @@
+            availability = region_availabilities.get(key, InstanceAvailability.NOT_AVAILABLE)
+            availability_offers.append(
+                InstanceOfferWithAvailability(**offer.dict(), availability=availability)
+            )
+
+        return availability_offers
+
+    def get_instance_state(self, instance_id: str, region: str) -> InstanceState:
+        instance = self.api_client.client.instances.get_by_id(instance_id)
+
+        # TODO 404 NOT_FOUND = "not_found"
+        # TODO other statuses
+
+        state_map = {
+            "provisioning": InstanceState.PROVISIONING,
+            "running": InstanceState.RUNNING,
+            "offline": InstanceState.STOPPED,
+        }
+        status = state_map.get(instance.status, InstanceState.NOT_FOUND)
+        return status
+
+    def run_job(
+        self,
+        run: Run,
+        job: Job,
+        instance_offer: InstanceOfferWithAvailability,
+        project_ssh_public_key: str,
+        project_ssh_private_key: str,
+    ) -> LaunchedInstanceInfo:
+        ssh_ids = []
+        ssh_ids.append(
+            self.api_client.get_or_create_ssh_key(
+                name=f"dstack-{job.job_spec.job_name}.key",
+                public_key=run.run_spec.ssh_key_pub.strip(),
+            )
+        )
+        ssh_ids.append(
+            self.api_client.get_or_create_ssh_key(
+                name=f"dstack-{job.job_spec.job_name}.key",
+                public_key=project_ssh_public_key.strip(),
+            )
+        )
+
+        commands = get_shim_commands(
+            backend=BackendType.DATACRUNCH,
+            image_name=job.job_spec.image_name,
+            authorized_keys=[
+                run.run_spec.ssh_key_pub.strip(),
+                project_ssh_public_key.strip(),
+            ],
+            registry_auth_required=job.job_spec.registry_auth is not None,
+        )
+
+        startup_script = " ".join(["sh", "-c", " && ".join(commands)])
+        startup_script_ids = self.api_client.get_or_create_startup_scrpit(
+            name=f"dstack-{job.job_spec.job_name}.sh", script=startup_script
+        )
+
+        name = job.job_spec.job_name
+
+        instance = self.api_client.deploy_instance(
+            instance_type=instance_offer.instance.name,
+            ssh_key_ids=ssh_ids,
+            startup_script_id=startup_script_ids,
+            hostname=name,
+            description=name,
+            image=job.job_spec.image_name,
+        )
+
+        launched_instance = LaunchedInstanceInfo(
+            instance_id=instance.id,
+            ip_address=instance.ip,
+            region=instance.location,
+            ssh_port=22,
+            username="root",
+            dockerized=False,
+            backend_data=None,
+        )
+
+        return launched_instance
+
+    def terminate_instance(
+        self, instance_id: str, region: str, backend_data: Optional[str] = None
+    ):
+        self.api_client.delete_instance(instance_id)
```

Comment:
`compute.terminate_instance` must not raise an error, if an instance was not found. Is that the case there?